src/utilities/useEnv.ts

```diff
@@ -0,0 +1,21 @@
+import { useStore } from '@/store/store'
+type EnvVars = {
+  KUMA_DOCS_URL: string
+  KUMA_UTM_QUERY_PARAMS: string
+  KUMA_FEEDBACK_URL: string
+}
+export const useEnv = () => {
+  const store = useStore()
+  const docsVersion = store.getters['config/getKumaDocsVersion']
+  const env: EnvVars = {
+    KUMA_DOCS_URL: `${import.meta.env.VITE_DOCS_BASE_URL}/${docsVersion}/`,
```

Comment:
**Question**: Why is this variable not created and stored in the store? In my opinion, this is where all computation of some application-wide stored data should happen.

This utility (i.e. `useEnv`) is effectively a secondary store for static/computed data. However, the store we create using Vuex already serves that purpose. Why do we need this store on top of it?

Since this store accesses the Vuex store using a composable (`useStore` wraps the Vuex `useStore` composable), it cannot be used in a context where the Vuex store can’t be used.

On a side note, I also think the name is slightly misleading. `useEnv` implies you’re accessing environment variables and while that’s certainly part of what the utility does, it does more than it — stuff environment variables usually don’t. I’d rather not muddle the waters and refer to environment variables in a utility that does more than access environment variables.

Reply:
1. We are moving away from Vuex.
2. We are moving away from these values being "user data". They are environment variables not API exposed user data. No reason to use all the previous code to manage runtime state when that state never changes.
3. Whether we do the computation as a build step or a runtime step, to be honest I don't really mind. It's just easier to do this here for now so we can move on and then begin to pull in all the other changes. This PR was meant to be "add a button" not "change how we pass in 'burnt in' environment variables". As it stands now, we achieve the "add a button" so we can merge this but also begin to prepare the way for the "change how we pass in burnt in environment variables"  which is happening upstream.
4. Naming-wise, I really don't want the name of our build tooling and other related things tightly coupled to our application.

Reply:
> We are moving away from Vuex.

That is news to me. When was this decided and by who and why?

> We are moving away from these values being "user data". They are environment variables not API exposed user data. No reason to use all the previous code to manage runtime state when that state never changes.

I’m not sure I understand this argument. All this data *is* runtime state and while it might never change that doesn’t mean it shouldn’t be stored in a Vuex store. Just because Vuex can recompute state in terms of Vue reactivity doesn’t mean this adds an unnecessary burden on storing static information in it.

State doesn’t immediately imply "stuff that’s constantly being recomputed". 

> Whether we do the computation as a build step or a runtime step, to be honest I don't really mind.

This computation doesn’t happen at build time in either approach.

> It's just easier to do this here for now so we can move on

It’s just as easy to do this in the Vuex store.

> and then begin to pull in all the other changes.

What do you mean by this?

> "change how we pass in burnt in environment variables" which is happening upstream.

I’m wholly unaware what you mean by this. What is "happening upstream"?

> Naming-wise, I really don't want the name of our build tooling and other related things tightly coupled to our application.

That’s an argument for introducing a utility that abstracts access to environment variable, but this utility does more than access environment variables. It assembles computed state. I think our respective definition of environment variables doesn’t match.

---

What I really like answered, however, is

> **Why** is this variable not created and stored in the store? In my opinion, this is where all computation of some application-wide stored data should happen.

What is the problem with the current approach of storing and accessing application state and what is it that you intend to change about it?

Reply:
Honestly, I just want to add a button 😆 🤷